commandapi-platforms/commandapi-paper/commandapi-paper-nms/commandapi-paper-nms-common/src/main/java/dev/jorel/commandapi/nms/PaperNMS_Common.java

```diff
@@ -0,0 +1,30 @@
+package dev.jorel.commandapi.nms;
+
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import dev.jorel.commandapi.CommandAPIPaper;
+import dev.jorel.commandapi.preprocessor.Unimplemented;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import net.kyori.adventure.text.serializer.gson.GsonComponentSerializer;
+import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.commands.arguments.MessageArgument;
+
+import static dev.jorel.commandapi.preprocessor.Unimplemented.REASON.VERSION_SPECIFIC_IMPLEMENTATION;
+
+public abstract class PaperNMS_Common extends CommandAPIPaper<CommandSourceStack> {
+
+	@Override
+	public final Component getChat(CommandContext<CommandSourceStack> cmdCtx, String key) throws CommandSyntaxException {
+		return GsonComponentSerializer.gson().deserialize(net.minecraft.network.chat.Component.Serializer.toJson(MessageArgument.getMessage(cmdCtx, key)));
+	}
```

Comment:
This method should be overriden in 1.20.5+ because `Serializer#toJson` was changed in that version to require the `COMMAND_BUILD_CONTEXT`. Currently, with this code:

```java
@Override
public void onEnable() {
    // Plugin startup logic
    CommandAPI.onLoad(new CommandAPIPaperConfig(this).verboseOutput(true));

    CommandAPI.onEnable();

    new CommandAPICommand("test")
            .withArguments(new ChatArgument("chat"))
            .executesPlayer(info -> {
                Component component = info.args().getUnchecked("chat");

                info.sender().sendMessage(component);
            })
            .register();
}
```

Running `/test Hello` in the console gives the following error, causing the command to fail.

```
Unexpected exception while parsing console command "test Hello"
org.bukkit.command.CommandException: Unhandled exception executing 'test Hello' in org.bukkit.craftbukkit.command.VanillaCommandWrapper(test)
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:175) ~[paper-mojangapi-1.21.1-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.CraftServer.dispatchCommand(CraftServer.java:1000) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at org.bukkit.craftbukkit.CraftServer.dispatchServerCommand(CraftServer.java:985) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at net.minecraft.server.dedicated.DedicatedServer.handleConsoleInputs(DedicatedServer.java:526) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:474) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1598) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1304) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:330) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.NoSuchMethodError: 'java.lang.String net.minecraft.network.chat.Component$Serializer.toJson(net.minecraft.network.chat.Component)'
        at BukkitTestPlugin-1.0-SNAPSHOT.jar/dev.jorel.commandapi.nms.PaperNMS_Common.getChat(PaperNMS_Common.java:19) ~[BukkitTestPlugin-1.0-SNAPSHOT.jar:?]
        at BukkitTestPlugin-1.0-SNAPSHOT.jar/dev.jorel.commandapi.arguments.ChatArgument.parseArgument(ChatArgument.java:80) ~[BukkitTestPlugin-1.0-SNAPSHOT.jar:?]
        at BukkitTestPlugin-1.0-SNAPSHOT.jar/dev.jorel.commandapi.arguments.ChatArgument.parseArgument(ChatArgument.java:46) ~[BukkitTestPlugin-1.0-SNAPSHOT.jar:?]
        at BukkitTestPlugin-1.0-SNAPSHOT.jar/dev.jorel.commandapi.CommandAPIHandler.parseArgument(CommandAPIHandler.java:319) ~[BukkitTestPlugin-1.0-SNAPSHOT.jar:?]
        at BukkitTestPlugin-1.0-SNAPSHOT.jar/dev.jorel.commandapi.CommandAPIHandler.argsToCommandArgs(CommandAPIHandler.java:291) ~[BukkitTestPlugin-1.0-SNAPSHOT.jar:?]
        at BukkitTestPlugin-1.0-SNAPSHOT.jar/dev.jorel.commandapi.CommandAPIHandler.lambda$generateCommand$0(CommandAPIHandler.java:195) ~[BukkitTestPlugin-1.0-SNAPSHOT.jar:?]
        at com.mojang.brigadier.context.ContextChain.runExecutable(ContextChain.java:73) ~[brigadier-1.3.10.jar:?]
        at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:31) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:19) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at net.minecraft.commands.execution.UnboundEntryAction.lambda$bind$0(UnboundEntryAction.java:8) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at net.minecraft.commands.execution.CommandQueueEntry.execute(CommandQueueEntry.java:5) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at net.minecraft.commands.execution.ExecutionContext.runCommandQueue(ExecutionContext.java:103) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at net.minecraft.commands.Commands.executeCommandInContext(Commands.java:443) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at net.minecraft.commands.Commands.performCommand(Commands.java:350) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at net.minecraft.commands.Commands.performCommand(Commands.java:337) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at net.minecraft.commands.Commands.performPrefixedCommand(Commands.java:327) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at org.bukkit.craftbukkit.command.VanillaCommandWrapper.execute(VanillaCommandWrapper.java:55) ~[paper-1.21.1.jar:1.21.1-41-fcedb49]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:165) ~[paper-mojangapi-1.21.1-R0.1-SNAPSHOT.jar:?]
        ... 8 more
```

Add the `@Overriden` annotation to this method matches the current implementation on `dev/dev` in [`NMS_Common`](https://github.com/JorelAli/CommandAPI/blob/c3ff409407ba18f3b0561051bdc99b1f89ca1678/commandapi-platforms/commandapi-bukkit/commandapi-bukkit-nms/commandapi-bukkit-nms-common/src/main/java/dev/jorel/commandapi/nms/NMS_Common.java#L310-L314). `dev/paper/restructure-experiment` currently [does this](https://github.com/JorelAli/CommandAPI/blob/d9c1f0d6a85cd94ab7d2d20682c12ad20684ea54/commandapi-platforms/commandapi-paper/commandapi-paper-nms/commandapi-paper-nms-common/src/main/java/dev/jorel/commandapi/nms/PaperNMS_Common.java#L18-L35), and I've verified that the Paper `ChatArgument` over there works on real servers.

Addtionally, `getChatComponent` can be implemented here, with the same exception of needed to be overridden in 1.20.5+.